WORKFLOW.rst

```diff
@@ -0,0 +1,216 @@
+   RPC OpenStack?)
+
+Development Workflow
+--------------------
+
+#. A bug is filed or a feature is proposed to |rpcos|
+
+#. After triage and prioritization, a developer creates a fix.
+
+#. The fix is pushed to a branch on that developer's fork.
+
+   This prevents another developer from being able to collaborate on that
+   change.
+
+#. The developer submits a pull request against ``master``. It's reviewed,
+   iterated upon, and eventually merged.
+
+#. If that PR is elligible to be backported, then the associated bug is
+   labeled as such and PRs are submitted to the stable branches.
+
+#. Stable PRs are then reviewed and merged.
+
+#. A developer will (eventually) close the original issue.
+
+Release Workflow
+----------------
+
+.. note::
+    This section is a bit unclear to me even now. It will need refinement.
+
+#. When targetted features, bugs, etc. are merged to the appropriate branch, a
+   Release Candidate (RC) tag is created, e.g., ``r11.0.0rc1``.
+
+#. Our QE and Support teams use that tag to test certain scenarios.
+
+#. We receive feedback and bug reports which are fixed. We then go back to
+   Step 1 until we have a version with which everyone is the least unhappy.
+   The most recent RC tag is also then the final tag, e.g., ``r11.0.0``.
+
+Our official release document (hidden away in a private wiki) states we should
+create a branch for a release before it is created, but recently when we
+released ``r11.1.0``, we created a separate branch ``kilo-11.1`` after we
+decided on the final acceptable ``r11.1.0`` RC tag.
+
+.. _proposed-workflow:
+
+RPC OpenStack's Proposed New Workflow
+=====================================
+
+Pre-Workflow
+------------
+
+#. Rename ``master`` to be |lsbnext|.
+
+#. Make |lsbnext| the "default" branch on GitHub.
+
+   This assures that all PRs will be targetted to |lsbnext| by default.
+
+.. note::
+
+    When a new stable release of OpenStack is created, |lsbnext| would then be
+    renamed to the codename of that release, e.g., ``liberty-next``.
+
+Development Workflow
+--------------------
+
+#. After a bug or feature is prioritized and assigned, the developer pushes a
+   fix to a branch on |rpcos|. [#]_
+
+#. A PR is sent to target the |lsbnext| branch (if the bug applies to
+   |lsbnext|, otherwise it is sent to the most recent branch it affects.
+
+#. All features and bugs are merged to |lsbnext| as appropriate.
+
+#. Bugs meeting backport criteria [#]_ are backported to the previous branches
+   as appropriate.
+
+#. Backport PRs are added to the original issue by the developers who create
+   them for easier tracking.
+
+   .. code::
+
+       {{ Issue Description }}
+
+       ----
+
+       - [ ] Original PR: https://github.com/rcbops/rpc-openstack/pull/{{number}}
+
+       - [ ] Backport to kilo: https://github.com/rcbpos/...
+
+       - [ ] ...
+
+   Using markdown checklists in GitHub will give others a quick view into the
+   status of each pull request.
```

Comment:
Maybe I misunderstood this section. As I read it the issue description would be used to track the branches to which the issues pertains. Currently tags are used to flag and issues for backports and that is useful be issues can be filtered based on a tag.